File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod=1
        for num in nums:
            prod=prod*num
        output=[]
        for i in range(len(nums)):
            if nums[i]!=0:
                output.append(prod//nums[i])
            else:
                pro=1
                for j in range(len(nums)):
                    if j!=i:
                        pro=pro*nums[j]
                output.append(pro)

        return output
        

File: Data Structures & Algorithms/jump-game/submission-0.java
class Solution {
    boolean []dp;
    public boolean canJump(int[] nums) {
        dp= new boolean[nums.length];
        if(nums.length==1){
            return true;
        }
        dp[0]=true;
        for(int i=0;i<nums.length;i++){
            if(!dp[i]){
                continue;
            }
            for(int j=1;j<=nums[i]&&i+j<nums.length;j++){
                dp[i+j]=true;
            }
        }
        return dp[nums.length-1];
    }
}
